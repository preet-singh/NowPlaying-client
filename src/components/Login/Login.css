.Login {
  display: flex;
  flex-direction: column;
  width: 30%;
  margin-left: auto;
  margin-right: auto;
  border: 1px solid black;
  justify-content: center;
  padding: 1em;
}