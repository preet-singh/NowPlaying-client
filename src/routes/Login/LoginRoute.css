@media screen and (min-width: 350px){

    .LoginRoute {
        background-color: rgb(8, 8, 8);
        min-height: 105vh;
        display: flex;
        justify-content: center;
    }

    .LoginRoute main form {
        border: none;
    }

    .LoginRoute main {
        width: 95%;
        display: flex;
        justify-content: center;
        margin-top: -100px;
    }
}