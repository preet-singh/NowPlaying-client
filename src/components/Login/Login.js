//Dependencies
import React from 'react';
import './Login.css';
import { Link } from 'react-router-dom';
import AuthApiService from '../../utils/auth-service';
import UserContext from '../../utils/context';

class Login extends React.Component{
  static contextType = UserContext;

  handleSubmit = e => {
    e.preventDefault();
    const { username, password } = e.target;
    AuthApiService.postLogin({ username, password })
    .then(res => {
      username.value = '';
      password.value = '';
      this.context.processLogin(res.authToken)
      this.props.onLoginSuccess()
    })
  }
  render() {
  return(
    <form className="Login" name="login" onSubmit={this.handleSubmit}>
      <legend>Login</legend>
      <label htmlFor="username-input" id="username-label" name="username-label">username:</label>
      <input type="text" id="username-input" name="username-input" />
      <label htmlFor="password-input" id="password-label" name="password-label">password:</label>
      <input type="text" id="password-input" name="password-input" />
      <button type="submit" value="submit">Submit</button>
      <Link to='/register'>
      <p className="sign-up">Don't have an account?</p>
      </Link>
    </form>
  )
  }
}

export default Login;