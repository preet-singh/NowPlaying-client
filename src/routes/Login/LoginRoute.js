//Dependencies
import React from 'react';

//Components
import Header from '../../components/Header/Header';
import Login from '../../components/Login/Login';


function LoginRoute(props) {
  return(
    <div className="LoginRoute">
      <Header />
      <main>
        <Login 
          history={props.history}
        />
      </main>
    </div>
  );

}

export default LoginRoute;