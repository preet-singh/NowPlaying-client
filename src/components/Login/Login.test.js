import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import Login from './Login';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('Login box testing', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Login />
            </BrowserRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the login form', () => {
        const wrapper = shallow(<Login />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

});