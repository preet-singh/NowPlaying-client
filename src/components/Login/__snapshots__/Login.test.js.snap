// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login box testing renders the login form 1`] = `
<form
  className="Login"
  name="login"
  onSubmit={[Function]}
>
  <legend>
    LOGIN
  </legend>
  <input
    id="username-input"
    name="username"
    onChange={[Function]}
    placeholder="ex. FlixMaster12"
    required={true}
    type="text"
  />
  <input
    autoComplete="on"
    id="password-input"
    name="password"
    onChange={[Function]}
    placeholder="********"
    required={true}
    type="password"
  />
  <button
    className="black-button"
    type="submit"
    value="submit"
  >
    Submit
  </button>
  <Link
    to="/register"
  >
    <p
      className="sign-up"
    >
      Don't have an account?
    </p>
  </Link>
</form>
`;
